src/lib/server/map_async.c

```diff
@@ -1168,6 +1197,35 @@ int map_list_mod_apply(REQUEST *request, vp_list_mod_t const *vlm)
 	}
 		goto finish;
 
+	case T_OP_PREPEND:
+	{
+		fr_cursor_t	to, from, from_reverse;
+		VALUE_PAIR	*vp_from, *vp, *vp_from_reverse = NULL;
+
+		vp_from = map_list_mod_to_vps(parent, vlm);
+		fr_assert(vp_from);
+
+		/*
+		 * First reverse the "from" list so they are in the
+		 * correct sequence to prepend to the "to" list
+		 */
```

Comment:
wouldn't it also be possible to prepend the first VP, and then just insert the subsequent VPs?  i.e.

prepend from_head, to_head
for the rest of from
   insert from into to
  next to